provider/bidengine/order.go

```diff
@@ -230,8 +231,13 @@ loop:
 
 func (o *order) shouldBid(group *dquery.Group) bool {
 
-	// does provider have required attributes?
-	if !group.MatchAttributes(o.session.Provider().Attributes) {
+	if !o.session.MatchAttributes(group.Requirements) {
+		o.log.Debug("unable to fulfill: required attributes not matched")
+		return false
+	}
+
+	// compares group requirements with provider attributes and required attributes
+	if !group.MatchAttributes(append(o.session.Provider().Attributes, o.session.RequiredAttributes()...)) {
```

Comment:
I don't think that this works.  this is `match(a,b+c) = match(a,b) && match(a,c)`, but we want `match(a,b) && match(c,a)`.